### ISSUE TITLE
`run_mlm.py` doesn't log perplexity to `wandb`

### ISSUE THREAD

**Issue by david-waterworth** [huggingface/transformers#23593]
### System Info

When I run `run_mlm.py`  all of the metrics which are created with names such as `train_loss` or `eval_loss` are logged to `wandb` reformatted as `train/loss` or `eval/loss`. However the eval perplexity is simply logged as `perplexity` and is not included in the wandb metrics

https://github.com/huggingface/transformers/blob/3658488ff77ff8d45101293e749263acf437f4d5/examples/pytorch/language-modeling/run_mlm.py#LL620C1-L635C46

The following is the output from `run_mlm.py`, you can observe that only metrics with a `eval_` prefix are included. I couldn't find in the code where the wandb login integration is or I'd dig further.

``` console
***** eval metrics *****
  epoch                   =       10.0
  eval_accuracy           =     0.9946
  eval_loss               =     0.0248
  eval_runtime            = 0:10:43.84
  eval_samples            =      62679
  eval_samples_per_second =     97.351
  eval_steps_per_second   =      3.043
  perplexity              =     1.0251
wandb: Waiting for W&B process to finish... (success).
wandb: / 0.130 MB of 0.130 MB uploaded (0.000 MB deduped)
wandb: Run history:
wandb:                  eval/accuracy ▁
wandb:                      eval/loss ▁
wandb:                   eval/runtime ▁
wandb:        eval/samples_per_second ▁
wandb:          eval/steps_per_second ▁
wandb:                    train/epoch ▁▂▂▃▃▄▄▅▅▆▆▇▇███
wandb:              train/global_step ▁▂▂▃▃▄▄▅▅▆▆▇▇███
wandb:            train/learning_rate █▇▇▆▆▅▅▄▄▃▃▂▂▁
wandb:                     train/loss █▇▆▅▅▄▃▃▂▂▂▂▁▁
wandb:               train/total_flos ▁
wandb:               train/train_loss ▁
wandb:            train/train_runtime ▁
wandb: train/train_samples_per_second ▁
wandb:   train/train_steps_per_second ▁
wandb: 
wandb: Run summary:
wandb:                  eval/accuracy 0.99464
wandb:                      eval/loss 0.02479
wandb:                   eval/runtime 643.8475
wandb:        eval/samples_per_second 97.351
wandb:          eval/steps_per_second 3.043
wandb:                    train/epoch 10.0
wandb:              train/global_step 16540
wandb:            train/learning_rate 0.0
wandb:                     train/loss 0.0568
wandb:               train/total_flos 3.677755155323775e+18
wandb:               train/train_loss 0.02581
wandb:            train/train_runtime 87644.5711
wandb: train/train_samples_per_second 144.98
wandb:   train/train_steps_per_second 0.189
```
I'm also not getting per-epoch eval metrics logged to wandb but I may have omitted a cmd line parameter.

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Run train_mlm with wandb enabled

### Expected behavior

Log `eval/perpleixity` to `wandb`
